use circleci/python:3.7-buster-node image and replace npm by yarn

willemarcel opened the pull request:
- It's a good practice to commit the `yarn.lock` file to assure the correct frontend package versions are instaled
- The `circleci/python:3.7-buster-browsers` is needed only when we need to run selenium tests, which needs a browser. On our case, I believe it's better to user the -node one, which already includes node and yarn packages

willemarcel commented:
I removed `- frontend/node_modules` from the `cached-dependencies` to avoid problems with previous dependencies installed by `npm`, but we can add it back in a next PR.

eternaltyro commented:
> I removed `- frontend/node_modules` from the `cached-dependencies` to avoid problems with previous dependencies installed by `npm`, but we can add it back in a next PR.

@willemarcel , this means that between one stage and a different stage of the CI process, the files will not be carried over. It will be as if `yarn` was not run at all, I think. @dakotabenjamin might have a better picture of the behaviour involved.

eternaltyro commented on .circleci/config.yml:
```diff
@@ -24,16 +23,15 @@ jobs:
         keys:
         - cached-dependencies
         paths:
-        - frontend/node_modules
```
Reconsider this.

willemarcel replied:
I'm not sure about it too. We can do a test without removing it.

willemarcel commented:
@eternaltyro I reverted the change on the `cached-dependencies` and replaced npm by yarn also on the installation of mapbox/cfn-config and cloudfriend

eternaltyro commented:
> @eternaltyro I reverted the change on the `cached-dependencies` and replaced npm by yarn also on the installation of mapbox/cfn-config and cloudfriend

@willemarcel thanks.

dakotabenjamin commented on .circleci/config.yml:
```diff
@@ -3,17 +3,16 @@ jobs:
   build:
     working_directory: /home/circleci/app
     docker:
-    - image: circleci/python:3.7-buster-browsers
```
if we remove `-browsers` will that affect the frontend tests? Or is that overwritten below in line 55?

willemarcel commented on .circleci/config.yml:
```diff
@@ -69,7 +68,7 @@ jobs:
         path: tests/backend/results
     - save_cache:
-        key: cached-dependencies
+        key: node-v1-{{ .Branch }}-{{ checksum "package-lock.json" }}
```
replace that line by `key: yarn-packages-v1-{{ .Branch }}-{{ checksum "yarn.lock" }}`

eternaltyro commented:
@dakotabenjamin looks good to merge to me (LGTMTM)
